### PR TITLE
Support `json` output (updated `config_object.json`)

### PR DESCRIPTION
**Pull request by YairSlobodin1**

### CODE REVIEW

**Review by kyorav — CHANGES_REQUESTED** [review]
Please update the command line support to add `'json` as a format.

File: pkg/io/confio/sg.go [hunk 1]
```diff
@@ -11,116 +11,242 @@ import (
...
+		return sgRemoteRef
+	}
+	ref := allocateRef()
+	sgRemoteRef := &vpcv1.SecurityGroupRuleRemoteSecurityGroupReference{
+		ID:   ref.ID,
+		CRN:  ref.CRN,
+		Href: ref.Href,
+		Name: utils.Ptr(name),
+	}
+	return sgRemoteRef
```
Comment (zivnevo) [inline pkg/io/confio/sg.go:54]: What about updating `nameToSGRemoteRef` with the newly created sgRemoteRef?

File: pkg/io/confio/sg.go [hunk 2]
```diff
@@ -11,116 +11,242 @@ import (
...
+}
+
+func makeSGRuleItem(nameToSGRemoteRef map[string]*vpcv1.SecurityGroupRuleRemoteSecurityGroupReference,
+	rule *ir.SGRule, i int) vpcv1.SecurityGroupRuleIntf {
 	iPVersion := utils.Ptr(ipv4Const)
 	direction := direction(rule.Direction)
+	cidrAll := ipblock.CidrAll
+	local := &vpcv1.SecurityGroupRuleLocal{
+		CIDRBlock: &cidrAll,
+	}
```
Comment (zivnevo) [inline pkg/io/confio/sg.go:83]: To be more consistent (and avoid code dup), also declare and assign remote variable here.

File: pkg/io/confio/sg.go [hunk 3]
```diff
@@ -11,116 +11,242 @@ import (
...
+		for j := range model.EndpointGWList[i].SecurityGroups {
+			sgID := model.EndpointGWList[i].SecurityGroups[j].ID
+			endpointGatewayID := model.EndpointGWList[i].ID
+			findAndDeleteTargetFromSG(model, idToSGIndex[*sgID], endpointGatewayID)
+		}
+		model.EndpointGWList[i].SecurityGroups = []vpcv1.SecurityGroupReference{sgRef}
 	}
+}
+
+func updateSG(model *configModel.ResourcesContainerModel, collection *ir.SGCollection) error {
```
Comment (zivnevo) [inline pkg/io/confio/sg.go:239]: The function never returns an error, so no need for return value.

**Review by zivnevo — CHANGES_REQUESTED** [review]

File: pkg/io/confio/conversions.go [hunk 4]
```diff
@@ -33,14 +33,14 @@ type refData struct {
 	Href *string
 }
 
-var globalIndex int = 0
+var GlobalIndex int = 0
```
[the remark this answers is not in the document]
Comment (YairSlobodin1) [reply pkg/io/confio/conversions.go:36]: for making test results deterministic

File: pkg/io/confio/sg.go [hunk 5]
```diff
@@ -11,116 +11,242 @@ import (
...
+const ResourceTypeEndpointGateway = "endpoint_gateway"
+
+func deleteTargetFromSG(model *configModel.ResourcesContainerModel, sgIndex, targetIndex int) {
+	part1 := model.SecurityGroupList[sgIndex].Targets[:targetIndex]
+	if targetIndex == len(model.SecurityGroupList[sgIndex].Targets)-1 {
+		model.SecurityGroupList[sgIndex].Targets = part1
+	} else {
+		part2 := model.SecurityGroupList[sgIndex].Targets[targetIndex+1:]
+		model.SecurityGroupList[sgIndex].Targets = append(part1, part2...) //nolint:gocritic // append result is assigned to the same slice
+	}
```
[the remark this answers is not in the document]
Comment (YairSlobodin1) [reply pkg/io/confio/sg.go:30]: `slices.Delete` replaces the targets with `null`. Is it good for us?

File: pkg/io/confio/sg.go (hunk 2 above)
Comment (YairSlobodin1) [reply pkg/io/confio/sg.go:83]: `local` should always be `0.0.0.0/0`. `sgRemote` returns the remote of the specific rule.

**Review by kyorav — APPROVED** [review]
ok by me

File: pkg/io/confio/conversions.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (zivnevo) [reply pkg/io/confio/conversions.go:34]: I don't get it. It's never used outside of this package

File: pkg/io/confio/acl.go [hunk 6]
```diff
@@ -102,56 +111,44 @@ func makeACLItem(acl *ir.ACL, subnet *vpcv1.SubnetReference) *configModel.Networ
...
-	resourceGroup := model.SubnetList[0].ResourceGroup
-	for _, subnetName := range collection.SortedACLSubnets("") {
-		acl := collection.ACLs[ir.VpcFromScopedResource(subnetName)][subnetName]
-		if acl == nil {
-			continue
-		}
-		subnetIndex := findSubnet(model, subnetName)
-		subnet := model.SubnetList[subnetIndex]
+	for i := range model.SubnetList {
+		vpc := model.SubnetList[i].VPC
```
[the remark this answers is not in the document]
Comment (zivnevo) [reply pkg/io/confio/acl.go]: Sorry, I missed the fact that `model.SubnetList` is a slice of pointers.
In that case, `for _, subnet := range model.SubnetList {` is preferred

File: pkg/io/confio/sg.go (hunk 2 above)
Comment (zivnevo) [reply pkg/io/confio/sg.go:84]: I meant something like `remote := sgRemote(nameToSGRemoteRef, rule)`, then use `remote` in each case of the switch.

File: pkg/io/confio/sg.go (hunk 5 above)
[the remark this answers is not in the document]
Comment (zivnevo) [reply pkg/io/confio/sg.go:30]: This should work:
```golang
model.SecurityGroupList[sgIndex].Targets = slices.Delete(model.SecurityGroupList[sgIndex].Targets, targetIndex, targetIndex+1)
```

File: pkg/io/confio/conversions.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (YairSlobodin1) [reply pkg/io/confio/conversions.go:34]: fixed

File: pkg/io/confio/sg.go [hunk 7]
```diff
@@ -7,126 +7,242 @@ package confio
...
+const ResourceTypeEndpointGateway = "endpoint_gateway"
+
+func findAndDeleteTargetFromSG(model *configModel.ResourcesContainerModel, sgIndex int, id *string) {
+	for i := range model.SecurityGroupList[sgIndex].Targets {
+		if target, ok := model.SecurityGroupList[sgIndex].Targets[i].(*vpcv1.SecurityGroupTargetReference); ok && *target.ID == *id {
+			model.SecurityGroupList[sgIndex].Targets = slices.Delete(model.SecurityGroupList[sgIndex].Targets, i, i+1) // deleteTargetFromSG
+			break
+		}
+	}
+}
```
Comment (zivnevo) [inline pkg/io/confio/sg.go:31]:
```suggestion
func findAndDeleteTargetFromSG(model *configModel.ResourcesContainerModel, sgIndex int, id *string) {
	sg := model.SecurityGroupList[sgIndex]
	for i := range sg.Targets {
		if target, ok := sg.Targets[i].(*vpcv1.SecurityGroupTargetReference); ok && *target.ID == *id {
			sg.Targets = slices.Delete(sg.Targets, i, i+1) // deleteTargetFromSG
			break
		}
	}
}
```

**Review by zivnevo — APPROVED** [review]

**Pull request merged**